extensions/functions_string.yaml

```diff
@@ -649,3 +649,18 @@ scalar_functions:
           - value: "string"
           - value: i32
         return: "string"
+
+aggregate_functions:
+
+  -
+    name: string_agg
+    description: Concatenates a column of string values with a separator.
+    impls:
+      - args:
+          - value: "string"
+            name: "input"
+            description: "Column of string values."
+          - value: "string"
+            name: "separator"
+            description: "Separator for concatenated strings"
+        return: "string"
```

Comment:
Nit, and not sure why none of the linters are complaining about this, but there should probably be a newline at the end of the file.